### PR TITLE
Feat/pd 4241 stack upgrade

### PR DESCRIPTION
**Pull request by hieu-w**
## Motivation and Context

Jira Link:
https://toruslabs.atlassian.net/browse/PD-4241

## Description

- [x] Update eslint + other packages to v9 in torus cli
- [x] Update to using flat eslint config in all repos + torus cli packages

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project. (run lint)
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code requires a db migration.

### CODE REVIEW

File: packages/eslint-config-node/index.js [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
+        ...globals.node,
+        ...globals.mocha,
+        ...globals.commonjs,
+        ...globals.es2020,
       },
-    ],
-    "n/no-missing-import": 0,
-    "import/no-extraneous-dependencies": ["error", { devDependencies: ["**/*.test.ts", "**/*.test.tsx", "**/testHelpers.ts"] }],
+
+      ecmaVersion: 11,
```
Comment (chaitanyapotti) [inline packages/eslint-config-node/index.js:21]: let's bump our code to support es2024. node 22 fully supports it.
exported code (library generated) should support node 20. (which is es2023)
so, eslint ecmaVersion should be 15 here

Comment (hieu-w) [reply packages/eslint-config-node/index.js:21]: we also can put es2024 here, should it still be 15 or es2024?

File: packages/eslint-config-node/index.js [hunk 2]
```diff
@@ -1,32 +1,53 @@
...
+        {
+          js: "never",
+          ts: "never",
+        },
+      ],
+
+      "n/no-unsupported-features/es-syntax": [
+        "error",
+        {
+          version: ">=16",
```
Comment (chaitanyapotti) [inline packages/eslint-config-node/index.js:38]: need to update this version also

File: packages/eslint-config-typescript/index.js [hunk 3]
```diff
@@ -1,70 +1,111 @@
...
+
+/** @type {import('@typescript-eslint/utils').TSESLint.FlatConfig.ConfigFile} */
+export default [
+  ...fixupConfigRules(compat.extends("problems")),
+  js.configs.recommended,
+  ...tseslint.configs.recommended,
+  eslintPluginPrettierRecommended,
+  importPlugin.flatConfigs.recommended,
+  pluginPromise.configs["flat/recommended"],
+  mochaPlugin.configs.flat.recommended,
```
Comment (chaitanyapotti) [inline packages/eslint-config-typescript/index.js:34]: add vitest here

File: packages/torus-scripts/package.json [hunk 4]
```diff
@@ -38,21 +38,21 @@
     "process": "^0.11.10",
     "release-it": "^17.10.0",
     "rimraf": "^6.0.1",
-    "rollup": "^4.24.4",
+    "rollup": "^4.28.1",
     "rxjs": "^7.8.1",
     "stream-browserify": "^3.0.0",
     "stream-http": "^3.2.0",
-    "typescript": "^5.6.3",
+    "typescript": "^5.7.2",
```
Comment (chaitanyapotti) [inline packages/torus-scripts/package.json:45]: use ~5.6.3 to keep compat with vue-tsc

File: packages/eslint-config-node/index.js (hunk 2 above)
Comment (hieu-w) [reply packages/eslint-config-node/index.js:38]: Updated

File: packages/eslint-config-typescript/index.js (hunk 3 above)
Comment (hieu-w) [reply packages/eslint-config-typescript/index.js]: Updated

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**